snuba/pipeline/processors.py

```diff
@@ -1,25 +1,74 @@
-import sentry_sdk
+from typing import Callable, Sequence
 
+import sentry_sdk
+from snuba.clickhouse.processors import QueryProcessor
 from snuba.datasets.entities.factory import get_entity
 from snuba.datasets.plans.query_plan import ClickhouseQueryPlan
-from snuba.request import Request
+from snuba.query.logical import Query as LogicalQuery
+from snuba.request.request_settings import RequestSettings
 
 
-def _execute_query_plan_processors(
-    query_plan: ClickhouseQueryPlan, request: Request
+def _execute_clickhouse_processors(
+    processors: Callable[[ClickhouseQueryPlan], Sequence[QueryProcessor]],
+    query_plan: ClickhouseQueryPlan,
+    settings: RequestSettings,
 ) -> None:
-    for clickhouse_processor in query_plan.plan_processors:
+    """
+    Executes the Clickhouse query processors for the query. These
+    are taken from the query plan.
+
+    When executing a query plan there are two sequences of query processors.
+    The first one is executed once per plan. The second is provided to the
+    execution strategy to be executed at every database query.
+    This function can be used in either case by customizing the sequence.
+    """
+    for clickhouse_processor in processors(query_plan):
         with sentry_sdk.start_span(
             description=type(clickhouse_processor).__name__, op="processor"
         ):
-            clickhouse_processor.process_query(query_plan.query, request.settings)
+            clickhouse_processor.process_query(query_plan.query, settings)
+
+
+def execute_pre_strategy_processors(
```

Comment:
How about `execute_plan_processors` or so  for naming consistency?

Reply:
ok.